feat(framework): upgrade esp idf framework (v5.x)

Hi, me again :-) i just split the topic for clarity, 
i tried to compile without any modification  in the code and couldnt success (i guess we got some different versions of arduino or something like that in the platformio setup since the are no versions of the packages defined.
the log states that the platform used is 2024.6.10 

here is the full log , maybe you can paste the log of a successful compilation to see any offending packages version.
```

 *  Exécution de la tâche : platformio run --environment arduino_nano_esp32 

Processing arduino_nano_esp32 (platform: espressif32; board: arduino_nano_esp32; framework: arduino)
----------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/arduino_nano_esp32.html
PLATFORM: Espressif 32 (2024.6.10) > Arduino Nano ESP32
HARDWARE: ESP32S3 240MHz, 320KB RAM, 16MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, dfu, esp-bridge, esp-builtin, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES: 
 - framework-arduinoespressif32 @ 3.0.1+sha.371d83e 
 - tool-dfuutil-arduino @ 1.11.0 
 - tool-esptoolpy @ 4.7.3 
 - tool-mklittlefs @ 3.2.0 
 - tool-riscv32-esp-elf-gdb @ 12.1.0+20221002 
 - tool-xtensa-esp-elf-gdb @ 12.1.0+20221002 
 - toolchain-riscv32-esp @ 12.2.0+20230208 
 - toolchain-xtensa-esp32s3 @ 12.2.0+20230208
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 31 compatible libraries
Scanning dependencies...
Dependency Graph
|-- RingBuffer @ 1.0.5
|-- FastCRC @ 1.41.0
|-- Time @ 1.6.1
|-- MQTT @ 2.5.2
|-- Adafruit BusIO @ 1.16.1
|-- RTClib @ 2.1.4
|-- Wire @ 2.0.0
|-- SPI @ 2.0.0
|-- ArduinoOTA @ 2.0.0
|-- WiFi @ 2.0.0
Building in release mode
Compiling .pio/build/arduino_nano_esp32/src/main.cpp.o
Compiling .pio/build/arduino_nano_esp32/src/message/MainStatusMessage.cpp.o
Compiling .pio/build/arduino_nano_esp32/src/state/DHWState.cpp.o
Compiling .pio/build/arduino_nano_esp32/src/state/HMIStateProxy.cpp.o
src/main.cpp: In function 'void setup()':
src/main.cpp:48:23: error: invalid conversion from 'uint8_t' {aka 'unsigned char'} to 'const esp_task_wdt_config_t*' [-fpermissive]
   48 |     esp_task_wdt_init(WATCHDOG_TIMEOUT_S, true);
      |                       ^~~~~~~~~~~~~~~~~~
      |                       |
      |                       uint8_t {aka unsigned char}
src/main.cpp:48:22: error: too many arguments to function 'esp_err_t esp_task_wdt_init(const esp_task_wdt_config_t*)'
   48 |     esp_task_wdt_init(WATCHDOG_TIMEOUT_S, true);
      |     ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from src/main.cpp:2:
/home/ouinouin/.platformio/packages/framework-arduinoespressif32/tools/esp32-arduino-libs/esp32s3/include/esp_system/include/esp_task_wdt.h:47:11: note: declared here
   47 | esp_err_t esp_task_wdt_init(const esp_task_wdt_config_t *config);
      |           ^~~~~~~~~~~~~~~~~
*** [.pio/build/arduino_nano_esp32/src/main.cpp.o] Error 1
Compiling .pio/build/arduino_nano_esp32/src/state/MainStateProxy.cpp.o
=================================================== [FAILED] Took 9.20 seconds ===================================================

Environment         Status    Duration
------------------  --------  ------------
arduino_nano_esp32  FAILED    00:00:09.197
============================================= 1 failed, 0 succeeded in 00:00:09.197 =============================================

 *  Arrêt du processus de terminal "platformio 'run', '--environment', 'arduino_nano_esp32'". Code de sortie : 1. 
 *  Le terminal sera réutilisé par les tâches, appuyez sur une touche pour le fermer. 
```

--- a/AquaMQTT/platformio.ini
+++ b/AquaMQTT/platformio.ini
@@ -18,7 +18,7 @@ lib_deps =
     googletest
 
 [env:arduino_nano_esp32]
-platform = espressif32@6.8.1
+platform = https://github.com/pioarduino/platform-espressif32/releases/download/51.03.05/platform-espressif32.zip
 board = arduino_nano_esp32
 framework = arduino
 test_framework = googletest

--- a/AquaMQTT/src/handler/OTA.cpp
+++ b/AquaMQTT/src/handler/OTA.cpp
@@ -1,6 +1,7 @@
 #include "handler/OTA.h"
 
 #include <ArduinoOTA.h>
+#include <esp_task_wdt.h>
 
 #include "config/Configuration.h"
 
@@ -36,6 +37,7 @@ void OTAHandler::setup()  // NOLINT(*-convert-member-functions-to-static)
             .onEnd([]() { Serial.println("\nEnd"); })
             .onProgress([](unsigned int progress, unsigned int total) {
                 Serial.printf("Progress: %u%%\r", (progress / (total / 100)));
+                esp_task_wdt_reset();
             })
             .onError([](ota_error_t error) {
                 Serial.printf("Error[%u]: ", error);

--- a/AquaMQTT/src/handler/RTC.cpp
+++ b/AquaMQTT/src/handler/RTC.cpp
@@ -21,43 +21,51 @@ void RTCHandler::setup()
 
 void RTCHandler::loop()
 {
-    if (timeStatus() != timeSet)
+    // do a ntp update if case time is not set or last ntp update is an hour ago
+    if (timeStatus() != timeSet || (millis() - mLastNTPUpdate) >= 3600000)
     {
-        if ((millis() - mLastNTPUpdate) >= 3600000)
+        mLastNTPUpdate = millis();
+        struct tm timeinfo
         {
-            mLastNTPUpdate = millis();
-            struct tm timeinfo
+        };
+
+        // success getting time via ntp
+        if (getLocalTime(&timeinfo, 5000))
+        {
+            // set local time in lib (PaulStoffregen/Time)
+            setTime(timeinfo.tm_hour,
+                    timeinfo.tm_min,
+                    timeinfo.tm_sec,
+                    timeinfo.tm_mday,
+                    timeinfo.tm_mon + 1,
+                    timeinfo.tm_year + 1900);
+
+            // store the new time to the rtc
+            if (mFoundRTC)
             {
-            };
-            if (getLocalTime(&timeinfo, 5000))
-            {
-                setTime(timeinfo.tm_hour,
-                        timeinfo.tm_min,
-                        timeinfo.tm_sec,
-                        timeinfo.tm_mday,
-                        timeinfo.tm_mon + 1,
-                        timeinfo.tm_year + 1900);
-                if (mFoundRTC)
-                {
-                    mRTC.adjust(DateTime(now()));
-                }
-            }
-            else if (mFoundRTC)
-            {
-                setTime(mRTC.now().unixtime());
+                mRTC.adjust(DateTime(now()));
             }
         }
+        // ntp update failed, but we have a hardware rtc
+        else if (mFoundRTC)
+        {
+            // set local time in lib from RTC
+            setTime(mRTC.now().unixtime());
+        }
     }
-    else if (config::OVERRIDE_TIME_AND_DATE_IN_MITM && (millis() - mLastStateUpdate) > 500)
+
+    // if time is known (either from RTC or NTP), then update the state
+    if (timeStatus() == timeSet && config::OVERRIDE_TIME_AND_DATE_IN_MITM && ((millis() - mLastStateUpdate) > 1000))
     {
-        mLastStateUpdate = now();
+        mLastStateUpdate  = millis();
+        time_t timeUpdate = now();
         aquamqtt::HMIStateProxy::getInstance().updateTime(
-                second(mLastStateUpdate),
-                minute(mLastStateUpdate),
-                hour(mLastStateUpdate),
-                day(mLastStateUpdate),
-                month(mLastStateUpdate),
-                year(mLastStateUpdate));
+                second(timeUpdate),
+                minute(timeUpdate),
+                hour(timeUpdate),
+                day(timeUpdate),
+                month(timeUpdate),
+                year(timeUpdate));
     }
 }
 

--- a/AquaMQTT/src/main.cpp
+++ b/AquaMQTT/src/main.cpp
@@ -21,6 +21,11 @@ OTAHandler     otaHandler;
 RTCHandler     rtcHandler;
 WifiHandler    wifiHandler;
 
+esp_task_wdt_config_t twdt_config = {
+    .timeout_ms     = WATCHDOG_TIMEOUT_S * 1000,
+    .idle_core_mask = (1 << configNUM_CORES) - 1,
+    .trigger_panic  = true,
+};
 
 void loop()
 {
@@ -44,7 +49,7 @@ void setup()
     Serial.println("REBOOT");
 
     // initialize watchdog
-    esp_task_wdt_init(WATCHDOG_TIMEOUT_S, true);
+    esp_task_wdt_init(&twdt_config);
     esp_task_wdt_add(nullptr);
 
     // setup wifi

